firefox-ios/firefox-ios-tests/Tests/XCUITests/TodayWidgetTests.swift

```diff
@@ -0,0 +1,443 @@
+        let firefoxWidgetSecureSearchButton = springboard
+            .buttons.matching(NSPredicate(format: "label CONTAINS[c] %@", "Private Tab")).element.exists
+        let firefoxCopiedLinkWidget = springboard.buttons
+            .matching(NSPredicate(format: "label CONTAINS[c] %@", "Copied Link")).element.exists
+        return firefoxWidgetButton || firefoxWidgetSecureSearchButton || firefoxCopiedLinkWidget
+    }
+
+    private func checkPresenceFifefoxWidget() -> Bool {
+        let maxSwipes = 3
+        var firefoxWidgetExists = false
+        var numberOfSwipes = 0
+
+        // Initial check for widget presence
+        if widgertExist() {
+            firefoxWidgetExists = true
+        } else {
+            // Perform swipe up until the widget is found or maxSwipes reached
+            while !springboard.buttons["Edit"].exists && numberOfSwipes < maxSwipes {
+                springboard.swipeUp()
+                if widgertExist() {
+                    firefoxWidgetExists = true
+                    break
+                }
+                numberOfSwipes += 1
+            }
+        }
+
+        return firefoxWidgetExists
+    }
+
+    private func removeFirefoxWidget() {
+        let maxSwipes = 3
+        var numberOfSwipes = 0
+
+        // Function to press and hold on a widget if it exists
+        func pressAndHoldWidget(matching label: String) {
+            let widget = springboard.buttons.matching(NSPredicate(format: "label CONTAINS[c] %@", label)).element
+            if widget.exists {
+                widget.press(forDuration: 1)
+            }
+        }
+
+        // Swipe up until the "Edit" button is visible or maxSwipes is reached
+        while !springboard.buttons["Edit"].exists && numberOfSwipes < maxSwipes {
+            springboard.swipeUp()
+            numberOfSwipes += 1
+        }
+
+        // Attempt to press and hold on any of the Firefox widget elements
+        pressAndHoldWidget(matching: "Firefox")
+        pressAndHoldWidget(matching: "Private Tab")
+        pressAndHoldWidget(matching: "Copied Link")
+
+        // Wait for and tap the remove button
+        mozWaitElementHittable(element: springboard.buttons[removeWidgetButton], timeout: 5)
+        springboard.buttons[removeWidgetButton].tap()
+
+        // Wait for the removal alert buttons to appear
+        mozWaitForElementToExist(springboard.alerts.buttons["Remove"])
+        mozWaitForElementToExist(springboard.alerts.buttons["Cancel"])
+
+        // Confirm widget removal
+        mozWaitElementHittable(element: springboard.alerts.buttons["Remove"], timeout: 5)
+        springboard.alerts.buttons["Remove"].tap()
+    }
+
+    private func checkFirefoxAvailablesWidgets() {
+        let maxWidgetCount = 5
+        var widgetCount = maxWidgetCount
+
+        // Check existence of widgets by swiping left through the widget list
+        XCTAssertTrue(springboard.staticTexts["Quick Actions"].exists, "Quick Actions widget not found.")
```

Comment:
Could we use `mozWaitForElementToExist` instead? We have had issue `XCTAssertTrue` to test the existence of elements because some elements don't show up right away.

Reply:
got it, yes we can use `mozWaitForElementToExist`